_overviews/scala3-book/first-look-at-types.md

```diff
@@ -0,0 +1,255 @@
+
+| Data Type     | Possible Values |
+| ------------- | --------------- |
+| Boolean       | `true` or `false` |
+| Byte          | 8-bit signed two’s complement integer (-2^7 to 2^7-1, inclusive)<br/>-128 to 127   |
+| Short         | 16-bit signed two’s complement integer (-2^15 to 2^15-1, inclusive)<br/>-32,768 to 32,767
+| Int           | 32-bit two’s complement integer (-2^31 to 2^31-1, inclusive)<br/>-2,147,483,648 to 2,147,483,647 |
+| Long          | 64-bit two’s complement integer (-2^63 to 2^63-1, inclusive)<br/>(-2^63 to 2^63-1, inclusive)   |
+| Float         | 32-bit IEEE 754 single-precision float<br/>1.40129846432481707e-45 to 3.40282346638528860e+38 |
+| Double        | 64-bit IEEE 754 double-precision float<br/>4.94065645841246544e-324d to 1.79769313486231570e+308d |
+| Char          | 16-bit unsigned Unicode character (0 to 2^16-1, inclusive)<br/>0 to 65,535 |
+| String        | a sequence of `Char` |
+
+
+
+## `BigInt` and `BigDecimal`
+
+When you need really large numbers, use the `BigInt` and `BigDecimal` types:
+
+```scala
+var a = BigInt(1_234_567_890_987_654_321L)
+var b = BigDecimal(123_456.789)
+```
+
+Where `Double` and `Float` are approximate decimal numbers, `BigDecimal` is used for precise arithmetic, such as when working with currency.
+
+A great thing about `BigInt` and `BigDecimal` is that they support all the operators you’re used to using with numeric types:
+
+```scala
+val b = BigInt(1234567890)   // scala.math.BigInt = 1234567890
+val c = b + b                // scala.math.BigInt = 2469135780
+val d = b * b                // scala.math.BigInt = 1524157875019052100
+```
+
+
+
+## Two notes about strings
+
+Scala strings are similar to Java strings, but they have two great additional features:
+
+- They support string interpolation
+- It’s easy to create multiline strings
+
+### String interpolation
+
+String interpolation provides a very readable way to use variables inside strings. For instance, given these three variables:
+
+```scala
+val firstName = "John"
+val mi = 'C'
+val lastName = "Doe"
+```
+
+You can combine those variables in a string like this:
+
+```scala
+println(s"Name: $firstName $mi $lastName")   // "Name: John C Doe"
+```
+
+Just precede the string with the letter `s`, and then put a `$` symbol before your variable names inside the string.
+
+To enclose expressions inside a string, put them in curly braces:
+
+```scala
+println(s"2 + 2 = ${2 + 2}")   // prints "2 + 2 = 4"
+val x = -1
+println(s"x.abs = ${x.abs}")   // prints "x.abs = 1"
+```
+
+
+#### Other interpolators
+
+The `s` that you place before the string is just one possible interpolator. If you use an `f` instead of an `s`, you can use `printf`-style formatting in the string. Furthermore, because `s` and `f` are really just methods, you can write your own interpolators, such as creating a `sql` interpolator for use in a database library. For more details, see this [String Interpolation page][interpolation].
```

Comment:
I think this is a bit too optimistic because it is not so easy for a beginner to define its own string interpolator, and the `sql` example is really complicated.

```suggestion
The `s` that you place before the string is just one possible interpolator. If you use an `f` instead of an `s`, you can use `printf`-style formatting syntax in the string. Furthermore, a string interpolator is a just special method and it is possible to define your own. For instance, some database libraries define the very powerful `sql` interpolator.
```